src/query/api/v1/handler/namespace/update.go

```diff
@@ -165,55 +183,82 @@ func validateUpdateRequest(req *admin.NamespaceUpdateRequest) error {
 func (h *UpdateHandler) Update(
 	updateReq *admin.NamespaceUpdateRequest,
 	opts handleroptions.ServiceOptions,
-) (nsproto.Registry, error) {
+) (nsproto.Registry, *xhttp.ParseError, error) {
 	var emptyReg = nsproto.Registry{}
 
 	store, err := h.client.Store(opts.KVOverrideOptions())
 	if err != nil {
-		return emptyReg, err
+		return emptyReg, nil, err
 	}
 
 	currentMetadata, version, err := Metadata(store)
 	if err != nil {
-		return emptyReg, err
+		return emptyReg, nil, err
 	}
 
-	updateID := ident.StringID(updateReq.Name)
-	newMDs := make([]namespace.Metadata, 0, len(currentMetadata))
+	newMetadata := make(map[string]namespace.Metadata)
```

Comment:
Worth initing with `len(currentMetadata)` or are there likely to be a lot of dupes?

Reply:
This API is called once in a blue moon at the read coordinator when reconfiguring a namespace, I'm all for optimized code but think any optimization here will be a burden more than help for maintenance/etc.